src/main/kotlin/frc/robot/subsystems/IntakeSubsystem.kt

```diff
@@ -0,0 +1,108 @@
+                      intakeMotorRID: Int,
+                      pitchMotorID: Int,
+) : SubsystemBase() {
+
+    val intakeMotorR = CANSparkMax(intakeMotorRID, CANSparkMaxLowLevel.MotorType.kBrushless)
+    val intakeMotorL = CANSparkMax(intakeMotorLID, CANSparkMaxLowLevel.MotorType.kBrushless)
+    val pitchMotor = CANSparkMax(pitchMotorID, CANSparkMaxLowLevel.MotorType.kBrushless)
+    var intakeMotors = MotorControllerGroup(intakeMotorL, intakeMotorR)
+
+    val pitchEncoder = pitchMotor.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle)
+    //var elbowPos = NetworkTableInstance.getDefault().getTable("Arm").getDoubleTopic("ElbowPositions")
+    // var desiredIntakePitch = NetworkTableInstance.getDefault().getTable("Intake").getDoubleTopic("IntakePositions").publish()
+    var desiredIntakeSpeed = NetworkTableInstance.getDefault().getTable("Intake").getDoubleTopic("IntakePositions").publish()
+    var desiredIntakePitch = NetworkTableInstance.getDefault().getTable("Intake").getDoubleTopic("IntakePositions").publish()
+    var intakePosition = NetworkTableInstance.getDefault().getTable("Intake").getDoubleTopic("IntakePositions").publish()
+
+
+
+    val pitchFFS = 0.0
+    val pitchFFG = 0.0
+    val pitchFFV = 0.0
+    val pitchFFA = 0.0
+    val FFVelocity = 0.0
+
+    var intakeFeedforward = ArmFeedforward(pitchFFS,pitchFFG, pitchFFV, pitchFFA)
+
+    val pitchP = 0.0
+    val pitchI = 0.0
+    val pitchD = 0.0
+
+    val intakePID = PIDController(pitchP,pitchI,pitchD)
+
+    val pitchEncoderPositionConversionFactor = 0.0
+    val pitchEncoderVelocityConversionFactor = 0.0
+
+    // limits for the position of the intake pitch
+    val maxIntakePosition = 2*Math.PI
+    val minIntakePosition = Math.PI
+
+    var targetIntakeSpeed = 0.0
+    var targetIntagePitch = 0.0
+
+    // when elbow pointing down --> 3pi/2
+    // when intake perpendicular to elbow --> 3pi/2
+    // offsets in revolutions
+    val elbowPosOffset = 0.0
+    val intakePosOffset = 0.0
+
+    //------------------------------------------------------------------------------------------------------------------------------------------------
+
+    init {
+        intakeMotorR.inverted = true
+        pitchEncoder.positionConversionFactor = pitchEncoderPositionConversionFactor
+        pitchEncoder.velocityConversionFactor = pitchEncoderVelocityConversionFactor
+    }
+
+    // radian pitch input
+    // targetPitch between pi and 2pi
+    fun setTargets(intakeSpeed: Double, intakePitch: Double){
+        targetIntakeSpeed = intakeSpeed
+        if (intakePitch <= minIntakePosition){
+            targetIntagePitch = minIntakePosition
+        }
+
+        else if (intakePitch >= maxIntakePosition){
+            targetIntagePitch = maxIntakePosition
+        }
+        else{
+            targetIntagePitch = (intakePitch + intakePosOffset)/(2*Math.PI)
+        }
+    }
+
+    //returns revs
+    fun getLevelPitch(elbowPos: Double) : Double{
```

Comment:
Refactor to be a utilty function that takes in both elbowPos (in radians) and the desired intakePitch to get what the value should be to get an "absolute" reading.

The idea is that if we set the target pitch to 0 degrees, the intake is always 0 degrees to the ground, regardless of what the elbow is at.